crates/proof-of-sql/src/sql/proof_plans/union_exec.rs

```diff
@@ -120,6 +120,7 @@ impl ProverEvaluate for UnionExec {
         let res = table_union(&inputs, alloc, self.schema.clone()).expect("Failed to union tables");
         builder.request_post_result_challenges(2);
         builder.produce_one_evaluation_length(res.num_rows());
+        builder.update_range_length(res.num_rows());
```

Comment:
Might be nice for `produce_one_evaluation_length` to internally/automatically call `update_range_length`. That way, it's more failsafe.

Reply:
@JayWhite2357 `update_range_length` is really only used for the few situations where `range_length` can potentially increase, mostly multi-table nodes (joins and unions) as well as range checks. So after the split `one_eval_lengths` and `range_length` no longer have much to do with each other.